### PR TITLE
Dianedev

### PR DESCRIPTION
**Pull request by dvanetten03**
Fixed db migrate file so it shows up.
Added name column to users.rb.
Add password_digest to users table.
Add password encryption, including bcrypt to Gemfile.
Correct spelling and add developer notes to README file.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,6 +1,49 @@
...
+  end
+
+  def show
+     @user = User.find(params[:id])
+  end
+
+  # Only users and admins can update user's profile
+  def update
+    if current_user.update_attributes(user_params)
+      redirect_to edit_user_registration_path
```
Comment (kenmazaika) [inline app/controllers/users_controller.rb]: This looks like it is missing an `end`.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,6 +1,49 @@
...
+  private
+    # Using a private method to encapsulate the permissable parameters
+
+    # Finds the User with the ID stored in the session with the key 
+    # :current_user_id. Logging in sets the session value and logging
+    # out removes it.
+    def current_user
+      @current_user || session[:current_user_id] &&
+        User.find_by(id: session[:current_user_id])
+      end
```
Comment (kenmazaika) [inline app/controllers/users_controller.rb]: This looks like it has an extra end.   Also, `current_user` should be populated by devise, we shouldn't need to build it.

File: app/models/user.rb [hunk 3]
```diff
@@ -7,4 +7,11 @@ class User < ActiveRecord::Base
 
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+  
+  before_save { self.email = email.downcase }
+  validates :name,  presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 }, format: { with: VALID_EMAIL_REGEX }, uniqueness: { case_sensitive: false }
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }
```
Comment (kenmazaika) [inline app/models/user.rb]: `devise :validatable` (line 9) should be doing most of this.  I don't think all of this is needed.

File: db/migrate/20160402015808_alter_sections_add_description_column.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 class AlterSectionsAddDescriptionColumn < ActiveRecord::Migration
   def change
-    add_column :sections, :description, :string
+    add_column :sections, :description, :string, null: false
```
Comment (kenmazaika) [inline db/migrate/20160402015808_alter_sections_add_description_column.rb]: We shouldn't modify existing migrations.  Only add new ones to modify the database.

File: lib/tasks/customdata.rake [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace :users do
+  
+  desc "This task helps to create admins"
+  task :admin, [:email, :password]
+end
+
+
+def add_Admin(email, password)
+  User.create(email: email, password: password, password_confirmation: password, admin: true)
```
Comment (kenmazaika) [inline lib/tasks/customdata.rake]: I believe @amarkpark added this in #65

File: spec/controllers/users_controller_spec.rb [hunk 6]
```diff
@@ -2,6 +2,7 @@
 
 RSpec.describe UsersController, type: :controller do
 
+
```
Comment (kenmazaika) [inline spec/controllers/users_controller_spec.rb]: We added a lot to the users controller, we should probably add some tests here.

**Comment by kenmazaika** [comment]
Just made a few comments.

**Comment by dvanetten03** [comment]
I made the changes you suggested directly to my branch, so they would be removed if the pull request was completed now. I will work on writing some tests.

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -1,6 +1,39 @@
...
+
+  def show
+     @user = User.find(params[:id])
+  end
+
+  # Only users and admins can update user's profile
+  def update
+    if current_user.update_attributes(user_params)
+      redirect_to edit_user_registration_path
+  end
```
Comment (kenmazaika) [inline app/controllers/users_controller.rb]: Still looks like a missing end here.

Comment (kenmazaika) [inline app/controllers/users_controller.rb]: Looks like all the other suggestions are fixed.  Still looks like there's, a `def` and `if` and a single `end`, when there should be 2.  I think this should cause problems when triggering the action in the controller.

Comment (kenmazaika) [inline app/controllers/users_controller.rb]: I'm happy to merge this though, when this issue is fixed.

Comment (dvanetten03) [inline app/controllers/users_controller.rb]: finally got it...so sorry I went through that code a couple of times and could not find the extra 'end'.  Thanks for Ken's QA!  New version being updated and sent now.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (kenmazaika) [inline app/controllers/users_controller.rb]: Unless I'm missing something, I still see un matched `ends` here:

```
def update
     if current_user.update_attributes(user_params)
       redirect_to edit_user_registration_path
    end
# an end should be here I believe, but it seems to be missing.
```

Unless I'm going crazy, haha.

**Pull request merged**